lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -0,0 +1,241 @@
+    logger = get_logger("__logger__")
+    try:
+        logger.setLevel(level=request.param)
+    except AttributeError:
+        logger.setLevel(level=logging.ERROR)
+    yield logger
+    logger.setLevel(level=logging.NOTSET)
+
+
+@pytest.fixture()
+def mock_tar_no_warnings(monkeypatch):
+    def fake_run(*args, **kwargs):
+        if "--warning=none" in args[0]:
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout=b"No error after --warning=none", stderr=None
+            )
+        else:
+            return subprocess.CompletedProcess(
+                args, returncode=1, stdout=b"Some error running tar", stderr=None
+            )
+
+    monkeypatch.setattr(subprocess, "run", fake_run)
+
+
+@pytest.fixture()
+def mock_tar_failure(monkeypatch):
+    def fake_run(*args, **kwargs):
+        return subprocess.CompletedProcess(
+            args,
+            returncode=1,
+            stdout=b"Some error running tar command, empty tar creation failed",
+            stderr=None,
+        )
+
+    monkeypatch.setattr(subprocess, "run", fake_run)
+
+
+class TestCreateTar:
+    tm_params = {
+        "benchmark_run_dir": "",
+        "channel_prefix": "",
+        "tds_hostname": "test.hostname.com",
+        "tds_port": 4242,
+        "controller": "test.hostname.com",
+        "group": "",
+        "hostname": "test.hostname.com",
+        "label": None,
+        "tool_metadata": {"persistent": {}, "transient": {}},
+        "tools": [],
+    }
+
+    def test_create_tar(self, agent_setup, mock_tar, logger_fixture):
+        """Test create tar file"""
+        tm = ToolMeister(
+            pbench_install_dir=None,
+            tmp_dir=None,
+            tar_path="tar_path",
+            sysinfo_dump=None,
+            params=self.tm_params,
+            redis_server=None,
+            logger=None,
+        )
+        tmp_dir = agent_setup["tmp"]
+        cp = tm._create_tar(tmp_dir, pathlib.Path(tar_file))
+        assert cp.returncode == 0
+        assert cp.stdout == b""
+
+    def test_create_tar_ignore_warnings(
+        self, agent_setup, mock_tar_no_warnings, logger_fixture
+    ):
+        """Test creating tar with warning=none option specified"""
+        tm = ToolMeister(
+            pbench_install_dir=None,
+            tmp_dir=None,
+            tar_path="tar_path",
+            sysinfo_dump=None,
+            params=self.tm_params,
+            redis_server=None,
+            logger=logger_fixture,
+        )
+        tmp_dir = agent_setup["tmp"]
+
+        cp = tm._create_tar(tmp_dir, pathlib.Path(tar_file))
+        assert cp.returncode == 0
+        assert cp.stdout == b"No error after --warning=none"
+
+    @pytest.mark.parametrize("logger_fixture", [logging.WARNING], indirect=True)
+    def test_create_tar_failure(
+        self, agent_setup, mock_tar_failure, caplog, logger_fixture
+    ):
+        """Test if we can suppress the errors raised during the tar creation"""
```

Comment:
Doesn't this actually verify that, without `retry`, we *report* the failure and a non-zero return status? We're not "suppressing" anything here.